shared/src/main/scala/com/comcast/ip4s/MulticastSocketAddress.scala

```diff
@@ -48,6 +57,22 @@ object MulticastSocketAddress {
   def fromString6(value: String): Option[MulticastSocketAddress[MulticastJoin, Ipv6Address]] =
     fromStringGeneric(value, V6Pattern, Ipv6Address(_))
 
+  def anySourceFromString(value: String): Option[MulticastSocketAddress[AnySourceMulticastJoin, IpAddress]] =
+    anySourceFromString4(value) orElse anySourceFromString6(value)
+  def anySourceFromString6(value: String): Option[MulticastSocketAddress[AnySourceMulticastJoin, Ipv6Address]] =
+    fromStringGeneric(value, V6Pattern, Ipv6Address(_)).flatMap(_.asAsm)
+  def anySourceFromString4(value: String): Option[MulticastSocketAddress[AnySourceMulticastJoin, Ipv4Address]] =
+    fromStringGeneric(value, V6Pattern, Ipv4Address(_)).flatMap(_.asAsm)
+
+  def sourceSpecificFromString(value: String): Option[MulticastSocketAddress[SourceSpecificMulticastJoin, IpAddress]] =
+    sourceSpecificFromString4(value) orElse sourceSpecificFromString6(value)
+  def sourceSpecificFromString6(
+      value: String): Option[MulticastSocketAddress[SourceSpecificMulticastJoin, Ipv6Address]] =
+    fromStringGeneric(value, V6Pattern, Ipv6Address(_)).flatMap(_.asSsm)
+  def sourceSpecificFromString4(
+      value: String): Option[MulticastSocketAddress[SourceSpecificMulticastJoin, Ipv4Address]] =
+    fromStringGeneric(value, V6Pattern, Ipv4Address(_)).flatMap(_.asSsm)
```

Comment:
V6Pattern with V4Address here and above on line 65 are bugs. Instead of calling `fromStringGeneric` here, can we call `fromString4` and `fromString6`?